front/src/pages/item/[id].tsx

```diff
@@ -1,15 +1,37 @@
-import React from "react";
-import ItemView from "@/components/ItemView";
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import ItemView from '../../components/ItemView';
+
+interface RowData {
+  id: string;
+  name: string;
+  age: string;
+  level: string;
+  message: string;
+  important: boolean;
+  status: string;
+}
 
 const ItemPage: React.FC = () => {
-  return (
-    <>
-      <>
-        <ItemView/>
-      </>
-      
-    </>
-  );
-};
+  const [data, setData] = useState<RowData[]>([]);
+  const router = useRouter();
+  const { id } = router.query;
 
-export default ItemPage;
+  useEffect(() => {
+    if (id) {
+      fetch(`/api/get/${id}`)
+        .then((res) => {
+        return  res.json();
+        })
+        .then((result) => {
+          setData(result.data);
+        })
+        .catch((error) => {
+          console.error('Error fetching data:', error);
+        });
+    }
+  }, [id]);
+
+  return <ItemView data={data}/>;
```

Comment:
try setting a loading logic for fetching before sending data because it might take time and sends the data empty before even fetching.
```js
  return (loading ? "loading" : <ItemView data={data} />);
```
something like this might work